components/tikv_util/src/lib.rs

```diff
@@ -90,6 +92,26 @@ pub fn panic_mark_file_exists<P: AsRef<Path>>(data_dir: P) -> bool {
     file::file_exists(path)
 }
 
+// create a file with hole, to reserve space for TiKV.
+pub fn reserve_space_for_recover<P: AsRef<Path>>(data_dir: P, file_size: u64) {
+    let path = data_dir.as_ref().join(RESERVE_FILE);
+    if file_size == 0 {
+        file::delete_file_if_exist(path).unwrap();
+        file::sync_dir(data_dir).unwrap();
+    } else {
+        if file::file_exists(path.clone()) {
+            if file::get_file_size(path.clone()).unwrap() == file_size {
+                return;
+            }
+            file::delete_file_if_exist(path.clone()).unwrap();
+        }
+        let f = File::create(path).unwrap();
+        f.allocate(file_size).unwrap();
```

Comment:
I think we should not panic if cannot reserve enough space configured.

Reply:
If the capacity of disk  is so small that TiKV can not reserve enough space, you can adjust config to a suitable value.